src/snackbar.vue

```diff
@@ -6,11 +6,23 @@
 
 <script>
 export default {
-  props: ['displayOn'],
-  ready () {
+  props: {
+    displayOn: {
+      type: String,
+      required: true
+    },
+    eventSource: {
+      type: Object,
+      required: false,
+      default: function () {
+        return this.$root
+      }
+    }
+  },
+  mounted () {
     componentHandler.upgradeElement(this.$el, 'MaterialSnackbar')
-    this.$on(this.displayOn, (snackarConfig) => {
-      this.$el.MaterialSnackbar.showSnackbar(snackarConfig)
+    this.eventSource.$on(this.displayOn, (snackbarConfig) => {
```

Comment:
What is eventSource. This is also something I'll have to change. The snackback component could be better